### ISSUE TITLE
OS-dependend directory separator LayoutRender 

### ISSUE THREAD

**Issue by DivineMgn** [NLog/NLog#3791]
### Problem
When using FileTarget, properties such as archiveFileName and fileName must be generated according to the target OS (another directory delimiter "/" or "\") for proper interpretation.
At the moment, to solve this problem, it is necessary to dynamically download the version of nlog-config with the correct paths depending on the OS or other manipulations.

For example (identical configs with only other "log-dir" variable):
nlog.win.config
```xml
<nlog>
 <variable name="log-dir" value="${basedir}\log\"/>
  <targets>
 <target xsi:type="File" fileName="${log-dir}logs.log"
..
/>
  </targets>
...
</nlog>
```
nlog.unix.config
```xml
<nlog>
 <variable name="log-dir" value="${basedir}/log/"/>
  <targets>
 <target xsi:type="File" fileName="${log-dir}logs.log"
..
/>
  </targets>
...
</nlog>
```
### Solution example
It would be great if there was a pre-defined layout render (directory path separator or OSVersion marker) to solve this problem.
Version:
For example:
```csharp
[LayoutRenderer("ds")]
public class DirectorySeparatorLayoutRenderer : LayoutRenderer
{
        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
        {
            builder.Append(Path.DirectorySeparatorChar);
        }
}
```
 nlog.config
```xml
<nlog>
 <variable name="log-dir" value="${basedir}${ds}log"/>
  <targets>
 <target xsi:type="File" fileName="${log-dir}${ds}logs.log"
..
/>
  </targets>
...
</nlog>
```

**Comment by 304NotModified** [comment]
Can't you use forward slashes in Windows? AFAIK I n windows you could use both. 

But have to agree we should add this feature

**Comment by 304NotModified** [comment]
Anyway, you could also add the DS in your application with a few lines of code 
 https://github.com/nlog/nlog/wiki/How-to-write-a-custom-layout-renderer

**Comment by DivineMgn** [comment]
> Can't you use forward slashes in Windows? AFAIK I n windows you could use both.
Yes, its work on windows. But if make a mistake with separator for linux (use windows notation '\') env its create log files in current app directory with incorrect names.

**Comment by snakefoot** [comment]
`ds` should be `directory-seperator`. But I also don't understand why `/` is not used.

Alternative one can setup LogDir using [GDC](https://github.com/nlog/nlog/wiki/Gdc-Layout-Renderer) at runtime.

**Comment by 304NotModified** [comment]
I made it as `${dir-separator}`, as all NLog "directory" renderers are named with "dir".

**Comment by 304NotModified** [comment]
fixed by #3793!
